### DISCUSSION TITLE
KubernetesExecutor Set Request and Limit CPU/Memory

### DISCUSSION THREAD

**Discussion by alice101-dev** [apache/airflow#22717] [Q&A] [answered]
Hi, im use airflow helm with version 
CHART VERSION = 1.2.0
APP VERSION = 2.1.4
executor = KubernetesExecutor
GKE = 1.20.15

how to set  Request and Limit CPU/Memory from DAG file ?

**Comment by potiuk** [comment]
See: https://airflow.apache.org/docs/apache-airflow/stable/executor/kubernetes.html#pod-template-file

**Comment by alice101-dev** [comment, marked as answer]
already find it 

executor_config={
            "KubernetesExecutor": {"request_cpu": "200m",
                                   "limit_cpu": "400m",
                                   "request_memory": "128Mi",
                                   "limit_memory": "400Mi"}},

docs from https://airflow.apache.org/docs/apache-airflow/stable/executor/kubernetes.html#pod-template-file very bad

**Reply by joarobles** [reply]
where did you find this?

**Reply by skannan-maf** [reply]
200m for request_cpu?? What does that mean?

**Reply by andresitorusz** [reply]
@skannan-maf 200m in Kubernetes CPU config means 200 millicores, or 0.2 vCPU (CPU core).

So, "request_cpu": "200m" means 20% of a single CPU core. Read more about it [here](https://kubernetes.io/docs/tasks/configure-pod-container/assign-cpu-resource/#cpu-units).

**Reply by skannan-maf** [reply]
Thank you for clarifying!!
@enchant3dmango

**Comment by nehemiascr** [comment]
you can also do
```json
"KubernetesExecutor": {
        "resources": {
            "requests": {
                "memory": "256Gi",
                "cpu": "4"
            }
        }
    }
```

**Comment by ut0mt8** [comment]
I guess the proper way to do it is now:

````
      task = PythonOperator(
          task_id="task",
          python_callable=stuff,
          executor_config={
              "pod_override": k8s.V1Pod(
                  spec=k8s.V1PodSpec(
                      containers=[
                          k8s.V1Container(
                              name="base",
                              resources=k8s.V1ResourceRequirements(
                                  requests={
                                      "cpu": "100m",
                                      "memory": "256Mi"
                                  },
                                  limits={
                                      "memory": "420Mi"
                                  }
                              )
                          )
                      ]
                  )
              )
          }
      )

````
